source/config/featureFlagEnums.py

```diff
@@ -0,0 +1,70 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+"""
+Feature flag value enumerations.
+Some feature flags require feature specific options, this file defines those options.
+All feature flags enums should
+- inherit from DisplayStringEnum and implement _displayStringLabels (for the 'displayString' property)
+- have a 'DEFAULT' member.
+"""
+import enum
+import typing
+
+from utils.displayString import (
+	DisplayStringEnum
+)
+
+from typing_extensions import Protocol  # Python 3.8 adds native support
+
+_displayString = typing.TypeVar("_displayString", bound=DisplayStringEnum)
+
+
+class FeatureFlagEnumProtocol(Protocol):
+	""" All feature flags are expected to have a "DEFAULT" value.
+	"""
+	DEFAULT: enum.Enum  # Required enum member
+
+
+FlagValueEnum = typing.Union[FeatureFlagEnumProtocol, _displayString]
+
+
+class BoolFlag(DisplayStringEnum):
+	"""Generic logically bool feature flag.
+	The explicit DEFAULT option allows developers to differentiate between a value set that happens to be
+	the current default, and a value that has been returned to the "default" explicitly.
+	"""
+
+	@property
+	def _displayStringLabels(self):
+		# To prevent duplication, self.DEFAULT is not included here.
+		return {
+			# Translators: Label for an option in NVDA settings.
+			self.DISABLED: _("Disabled"),
+			# Translators: Label for an option in NVDA settings.
+			self.ENABLED: _("Enabled"),
+		}
+
+	DEFAULT = enum.auto()
+	DISABLED = enum.auto()
+	ENABLED = enum.auto()
+
+	def __bool__(self):
+		if self == BoolFlag.DEFAULT:
+			raise ValueError(
+				"Only ENABLED or DISABLED are valid bool values"
+				", DEFAULT must be combined with a 'behavior for default' to be Truthy or Falsy"
+			)
+		return self == BoolFlag.ENABLED
+
+
+def getAvailableEnums() -> typing.Generator[typing.Tuple[str, DisplayStringEnum], None, None]:

```

Comment:
Given you had to add an exception for `DisplayStringEnum` to this function already wouldn't it be simpler not to try to detect available enums, rather create a class decorator which would be used to register known implementations? I'm worried that someone may import here something for which these conditions matches and fails to notice.

Reply:
I'm hesitant to complicate it further. There are unit tests that ensure that all collected enums have a `DEFAULT` which should catch most similar cases. If there was an import that brought along an enum that inherited from `DisplayStringEnum` and had a `DEFAULT` member, then it still doesn't do any harm really. I want to keep it as simple as possible to add new feature flag config.